ipset.go

```diff
@@ -4,167 +4,36 @@
 
 package netaddr
 
-import "sort"
-
-// IPSet represents a set of IP addresses.
+// IPSetBuilder is an IPSet builder.
 //
 // The zero value is a valid value representing a set of no IPs.
 //
-// The Add and Remove methods add or remove IPs to/from the set.
-// Add methods should be called first, as a remove operation does
-// nothing on an empty set. Ranges may be fully, partially, or not
-// overlapping.
-type IPSet struct {
+// The Add and Remove methods add or remove IPs to/from the
+// set. Removals only affect the current membership of the set, so in
+// general Adds should be called first. Input ranges may overlap in
+// any way.
+type IPSetBuilder struct {
```

Comment:
regarding your name qualms: if this were called MutableIPSet, that would make it obvious that IPSet weren't mutable, and would make `RemoveFreePrefix` less gross (not being a method on a "builder" type)

Reply:
MutableIPSet suggests that it should have all the various accessors that IPSet has, in my mind. I'd rather avoid that and keep it as a pure builder type.

I'm leaning a little more to Josh's suggestion of moving RemoveFreePrefix over to IPSet, and just make it produce a new IPSet on use. Makes it a little more expensive to use, but it's not a hot path.